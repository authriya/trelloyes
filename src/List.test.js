import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('List component', () => {
    it('renders without crashing', () => {
        const object = [{
            id: "one",
            title: "onetwo",
            content: "onetwothree"
        }]
        const div = document.createElement('div');
        ReactDOM.render(<List header="list" cards={object}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const object = [{
            id: "one",
            title: "onetwo",
            content: "onetwothree"
        }]
        const tree = renderer
        .create(<List header="list" cards= {object}/> )
        .toJSON()
        expect(tree).toMatchSnapshot();
    });

})