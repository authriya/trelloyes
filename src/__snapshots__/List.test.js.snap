// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        onetwo
      </h3>
      <p>
        onetwothree
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
